src/utils.ts

```diff
@@ -514,6 +515,21 @@ export function hasAtomicOperators(doc: Document | Document[]): boolean {
   return keys.length > 0 && keys[0][0] === '$';
 }
 
+export function resolveTimeoutOptions<T extends Partial<TimeoutContextOptions>>(
+  client: MongoClient,
+  options?: T
+): Pick<
+  MongoClient['s']['options'],
+  'serverSelectionTimeoutMS' | 'socketTimeoutMS' | 'waitQueueTimeoutMS' | 'timeoutMS'
+> &
```

Comment:
What are you saying is odd?  That I pick the options from the client's options type?

Reply:
By picking the options you've made it so this will automatically change with any changes to the client options which can hide errors since you aren't introducing an incompatibility when the types change, by redeclaring as we have in `timeout.ts` the "two sources of truth" provides value by ensuring we compile our inputs against our expected outputs.

I think this is fine to leave as is, but the above is just why it is odd to me.

Reply:
That's intentional.  This function makes no assumptions about the types of each argument, it just plucks them if present.  The return type is just the ouptut options type with the added properties from the MongoClient.